README.md

```diff
@@ -43,120 +36,74 @@ Check the original technical report
-| framework         | string                             | Used framework (ex: PyTorch, TensorFlow).                                    |
-| version           | string                             | Framework version (some models require a specific version of the framework). |
-| model_handler     | string                             | Inference execution function.                                                |
-| model_src_url     | string                             | Url of the source code (ex: GitHub repo).                                    |
-| model_commit_hash | string                             | Hash value pointing to a specific version of the code.                       |
-| docker            | \[[Docker Object](#docker-object)] | Information for the deployment of the model in a docker instance.            |
-
-#### Docker Object
+| total_parameters        | integer | Total number of parameters.                                 |
+| on_disk_size_mb         | number  | The memory size on disk of the model artifact (MB).         |
+| ram_size_mb | number    | number  | The memory size in accelerator memory during inference (MB).|
+
+### Runtime
+
+| Field Name            | Type                               | Description                                                                              |
+|-----------------------|------------------------------------|------------------------------------------------------------------------------------------|
+| framework             | string                             | Used framework (ex: PyTorch, TensorFlow).                                                |
+| version               | string                             | Framework version (some models require a specific version of the framework).             |
+| model_asset        | [Asset Object](https://github.com/radiantearth/stac-spec/blob/master/collection-spec/collection-spec.md#asset-object)                        | Common Metadata Collection level asset object containing URI to the model file.   |
+| model_handler         | string                             | Inference execution function.                                                            |
+| source_code_url         | string                             | Url of the source code (ex: GitHub repo).                                                |
+| model_commit_hash     | string                             | Hash value pointing to a specific version of the code.                                   |
+| docker                | [Container](#container) | Information for the deployment of the model in a docker instance.                        |
+| batch_size_suggestion | number                             | A suggested batch size for a given compute instance type                                 |
+| hardware_suggestion   | str                                | A suggested cloud instance type or accelerator model |
```

Comment:
I think more explicit is better here for model users but it does but more burden on the maintainers to keep the enum up to date and exhaustive and publishers to know what to pick. I'm uncertain but will look at how others do this.

Reply:
I think inside the runtime is more relevant than inside the model. Some models are able to run using CPU-only, single-GPU or multi-GPU without modification to the code. The inputs/outputs usually also require to be pushed/pulled between the devices employed, so the hardware is not limited to the model. 

I agree that this definition might be required for some cases, but we must think of a way to allow the flexibility to indicate when a mostly is not strictly required to run with a specific device. In most cases (at least from my experience), it is more often the minimal memory required on the hardware that is the limiting factor rather than the specific hardware to run it.

Reply:
an example of aversion to docker/.devcontainers: https://github.com/microsoft/torchgeo/pull/1085#issuecomment-1416381540

many models on huggingface related to geospatial or other models don't have associated Dockerfiles.

Reply:
> The inputs/outputs usually also require to be pushed/pulled between the devices employed, so the hardware is not limited to the model.

TLDR, I think these sorts of hardware specific optimizations are going to be more common as it gets easier to export models with AOT, so it'd be valuable to get more specific about the kind of accelerator beyond CPU/GPU/TPU

after thinking about all the times the GPU type has mattered to me, I think I lean towards getting more specific than CPU/GPU/TPU. in my personal experience, it's been an issue 2 or 3 times when trying newer architectures on K80s vs newer NVIDIA architectures.

Also, I think models trained on TPUs, Apple Silicon, and models that make use of hardware specific optimizations are becoming more common.

See https://pytorch.org/blog/2/, a lot of the recent blog posts and presentations at Pytorch Conference 2023 focused on new hardware specific optimizations for different accelerators.

One specific example is that models with custom Triton Kernels are not as portable as models trained on GPUs without optimizations.  I tried to run inference with [the segment anything fast model](https://pytorch.org/blog/accelerating-generative-ai/) and ran into issues because the triton kernel was optimized for A100s instead of my 3090. 

> we must think of a way to allow the flexibility to indicate when it is not strictly required to run with a specific device

Maybe we introduce a boolean flag like `accelerator-constrained`. If True, the model strictly only runs on that hardware. 

If it is just inadvisable to run it on different hardware for performance reasons, that could be specified in the `hardware suggestion` which I'd like to rename to `hardware summary` to encourage more description about the hardware the model has been run on

Examples: "8 NVlinked A100s", "1 local 3090", "1 compute optimized 32Gb RAM AWS CPU instance", "1 AWS Sagemaker serverless endpoint with 8GB RAM and a concurrency of 20"

Another example, the Segment Anything Model decoder runs fine on the CPU. The encoder will run on the CPU, but is extremely slow, and since SAM is primarily useful in user interaction scenarios because of it's promptability, I think it'd be helpful to advise using a GPU in the `hardware summary`

Reply:
I added an initital required enum for the intended accelerator and a boolean flag for accelerator-constrained

```
#### Accelerator Enum

It is recommended to define `accelerator` with one of the following values:

- `amd64` models compatible with AMD or Intel CPUs (no hardware specific optimizations)
- `cuda` models compatible with NVIDIA GPUs
- `xla` models compiled with XLA. models trained on TPUs are typically compiled with XLA.
- `amd-rocm` models trained on AMD GPUs
- `intel-ipex-cpu` for models optimized with IPEX for Intel CPUs
- `intel-ipex-gpu` for models optimized with IPEX for Intel GPUs
- `macos-arm` for models trained on Apple Silicon
```

Reply:
> Maybe we introduce a boolean flag like accelerator-constrained. If True, the model strictly only runs on that hardware.

That is a good idea. 

I like the possibility to provide more details to help reproduce experiments or provide context, but at the same time, a summary makes it less machine-readable. Users are free to ignore the hardware-suggestion value, but it could be useful to have a somewhat parse-able set of devices.

I think the hardware/accelerator definition might evolve with further details. Therefore, I suggest we make it an `Accelerator Object` on its own, which then would contain `constrained`, `suggested` and so on.

Reply:
I think machine readability might only be useful if we enforce an enum for hardware. If we don't, model publishers will specify NVIDIA GPU series, AWS EC2 instances, Google VMs, Azure VMs, etc. using whatever case, spelling, hyphenation they want. I think hardware will mean different things to different folks. they might be using a managed service or they might be familiar with the actual GPU running the computation.

Since there are so many options when it comes to specific hardware, machine readability probably goes out the window with all the opportunities for variation with hardware?

I expect `hardware_summary` to be read carefully by the ML practitioner when deploying the model, and for other fields to constrain model search (`accelerator`, `accelerator_constrained`). I think with the current options for the `accelerator` enum, it covers most classes of compute architecture. except edge devices like phones, we should probably add these to the enum at some point.

I agree the definition could evolve in the future and can add these accelerator related fields to a new object.

Reply:
Good points. I am fine with the justification for a generic summary so we don't have to maintain the hardware enum. 

The fact that you mentioned model search with the accelerator reminded me that STAC suggests avoiding nested objects since those cannot be searched easily by the API. So, let's keep `accelerator-constrained` and `accelerator-summary` separate.

Reply:
cool cool! one last nit, I think we may want to stick to `hardware summary` since it sounds more all encompassing than `accelerator summary` and leaves room to include details on hardware besides the accelerator.